doc/problem/classification.md

```diff
@@ -1,29 +1,62 @@
+
+The researchers record the daily activities of each mouse, for example the time it spends eating, drinking, sleeping, and wondering around its habitats. Therefore, every behavioral features should be averaged to the same time period (one mouse day) for each mouse. For example, the food consumed variable at each timestamp will be aggregated to the average food consumption.
+
+Notice that the final dataset should be a clean and well formatted dataframe (in numpy array or pandas dataframe) aggregating the features of mice so that it can be directly used to train classification models.
+
+Besides, the detailed explanation for each variable and strain type might be needed for further interpretations of models.
 
 ## Methodology/ Approach Description
+1.Supervisied Classification:
+Supervised classification algorithms (logistic regression, random forest, KNN) will be used to detect the relationship between strain of mice and behavioral features. If we gain good model performance, we can conclude different mouse behaviors actually indicate different genetic differences. K-fold cross-validation might be used to tune model parameters, and a proportion of data would be used as test data to evaluate the model perfomance. Notice that we may manually manipulate so that the both the training data and the test data cover all strains of mice.
+
+
+2.Unsupervisied clustering
+In addition, we can use unsupervised machine learning models (e.g. K-means) to cluster the daily mouse activities into clusters that correspond to the genes. This means we will not use the given strain label, instead we will create new labels for mice purely based on its behaviors by clustering. This can highlight an even stronger relationship between genes and mouse behaviors. 
 
-One way to detect this relationship between genes and behaviors is to use supervised machine learning models, and see whether we can use mouse daily habit to predict its gene. If we can get high accuracy, we can conclude that there is an association between the genes and behaviors, or nature trumps nurture. In addition, we can use unsupervised machine learning models to cluster the daily mouse activities into clusters that correspond to the genes. This can highlight an even stronger relationship between genes and mouse behaviors. 
+
+The step after model fitting is to assess the important behavioral features in the classification and clustering models. A smaller set of feature space containing only top features might be used to gain better interpretations of the model.
 
 ## Testing Framework outline
+- The first step to test the reproducibility is to test the stability of classification models. Since we randomly split the dataset to be the test set and the training set, we can train and test the model over different seeds and plot the accuracies against different trials. We should also see if the important features are stable over different trials. 
+
```

Comment:
Since the purpose of unit testing is to ensure our function is doing the correcting thing, we can start by checking whether the output has the expected length, or whether it has the expected output. For example, since we are clustering strains, we know there should be a maximum of 16 unique strain numbers.
We should implement functions like `test_classification` and `pytest` and the `assert` function would be very useful.